lib/administrate/field/associative.rb

```diff
@@ -18,15 +38,39 @@ def associated_dashboard
       end
 
       def associated_class_name
-        options.fetch(:class_name, attribute.to_s.singularize.camelcase)
+        if option_given?(:class_name)
+          deprecated_option(:class_name)
+        else
+          self.class.associated_class_name(
+            resource.class,
+            attribute,
+          )
+        end
       end
 
       def primary_key
-        options.fetch(:primary_key, :id)
+        if option_given?(:primary_key)
+          deprecated_option(:primary_key)
+        else
+          :id
+        end
       end
 
       def foreign_key
-        options.fetch(:foreign_key, :"#{attribute}_id")
+        if option_given?(:foreign_key)
+          deprecated_option(:foreign_key)
+        else
+          self.class.foreign_key_for(resource.class, attribute)
+        end
+      end
+
+      def option_given?(name)
+        options.key?(name)
+      end
+
+      def deprecated_option(name)
```

Comment:
naming methods as `deprecated_...` seems like a nice reminder, but can you be more specific, like...?
`drop_in_v1__deprecated_option`

Reply:
No problem; the `deprecated` label is clear enough that someone can search for it throughout the codebase when we do decide to remove them. You can keep as-is.